tools/publishIsolate.ts

```diff
@@ -0,0 +1,56 @@
+import fs from 'fs';
+import path from 'path';
+import childProcess from 'child_process';
+import getNpmPackageVersion from 'get-npm-package-version';
+import { compareVersions } from 'compare-versions';
+
+async function main() {
+  const isolatePackageRoot = './dist/isolate';
+
+  const publishPackageList: { packageName: string; dir: string }[] = [];
+
+  // get directory listing
+  const files = fs.readdirSync(isolatePackageRoot);
+  for (const file of files) {
+    const filePath = path.join(isolatePackageRoot, file);
+    const stat = fs.statSync(filePath);
+    if (stat.isDirectory()) {
+      // get package.json
+      const packageJsonPath = path.join(filePath, 'package.json');
+      const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+      // get version
+      const version = getNpmPackageVersion(packageJson.name);
+
+      if (version === null) {
+        throw new Error(`Package ${packageJson.name} not found on npm`);
+      }
+
+      // compare version
+      if (compareVersions(version, packageJson.version) < 0) {
+        publishPackageList.push({ packageName: packageJson.name, dir: filePath });
+      }
+    }
+  }
+
+  // publish packages
+  // eslint-disable-next-line compat/compat
+  await Promise.all<void>(publishPackageList.map(publishPackage => new Promise((resolve) => {
+    const childProcessInstance = childProcess.spawn(
+      'yarn',
+      ['publish', publishPackage.dir]
+    );
+
+    childProcessInstance.stderr.on('data', (data) => {
+      throw new Error(data.toString());
+    });
+
+    childProcessInstance.on('close', (code) => {
```

Comment:
Can this process add some log and output?

Reply:
To do so, Promise all must be removed